### ISSUE TITLE
Compilation error on simple <= 0 test

### ISSUE THREAD

**Issue by jeanlaroche** [serge-sans-paille/pythran#827]
I'm encoutering a bug: the following code results in a compilation error:
```
import numpy as np
# pythran export testFunc(int,int)
def testFunc(inLeft, inRight):
    # nHop = int(round(1. * inLeft / inRight)) # This works
    nHop = int(np.round(1. * inLeft / inRight)) # This does NOT work
    # nHop = np.round(1. * inLeft / inRight).astype(int) # This does NOT work
    if nHop <= 0: return 0
    return 1
```

I'm using version 0.8.4post0 on MacOS 10.12.6

**Comment by serge-sans-paille** [comment]
I fail to repdouce the error... can you test from master version and provide me with the error log (if any ?)

**Comment by serge-sans-paille** [comment]
@jeanlaroche up?

**Comment by jeanlaroche** [comment]
Tried with a fresh clone of master. Still getting the error. Here's the compilation error:


   `In file included from /var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmp_81yA3.cpp:11:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/include/numpy/round.hpp:4:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/include/numpy/around.hpp:5:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/include/numpy/floor_divide.hpp:13:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/div.hpp:55:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/scalar/div.hpp:14:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/common/scalar/function/div.hpp:18:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/iround.hpp:44:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/scalar/iround.hpp:14:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/common/scalar/function/iround.hpp:20:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/round.hpp:57:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/simd/round.hpp:15:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/common/simd/function/round.hpp:30:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/nearbyint.hpp:43:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/simd/nearbyint.hpp:19:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/x86/sse2/simd/function/nearbyint.hpp:14:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/inearbyint.hpp:50:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/scalar/inearbyint.hpp:21:
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/x86/sse2/scalar/function/inearbyint.hpp:31:14: error: use of undeclared identifier '_mm_cvtsd_si64'
      return _mm_cvtsd_si64(_mm_set_sd(a0));
             ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/detail/dispatch/function/functor.hpp:76:43: note: in instantiation of member function 'boost::simd::ext::impl_inearbyint_<boost::simd::sse2_, boost::dispatch::scalar_<boost::dispatch::double_<double> > >::operator()' requested here
      return BOOST_DISPATCH_IMPL_TAG_CALL(Tag,Site,Args,args);
                                          ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/numpy/rint.hpp:20:14: note: in instantiation of function template specialization 'boost::dispatch::functor<boost::simd::tag::inearbyint_, boost::simd::sse4_1_>::operator()<const double &>' requested here
      return boost::simd::inearbyint(v);
             ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/types/numpy_nary_expr.hpp:27:12: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::numpy::wrapper::rint<double>' requested here
    return NUMPY_NARY_FUNC_SYM(std::forward<T>(args)...);
           ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/numpy/rint.hpp:30:38: note: expanded from macro 'NUMPY_NARY_FUNC_SYM'
#define NUMPY_NARY_FUNC_SYM wrapper::rint
                                     ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/numpy/around.hpp:20:12: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::numpy::functor::rint::operator()<const double &>' requested here
    return functor::rint{}(a);
           ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/numpy/around.hpp:56:35: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::numpy::around<double>' requested here
  DEFINE_FUNCTOR(pythonic::numpy, around);
                                  ^
/var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmp_81yA3.cpp:39:49: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::numpy::functor::around::operator()<double>' requested here
    if ((pythonic::__builtin__::functor::int_{}(pythonic::numpy::functor::round{}((pythonic::operator_::div((1.0 * inLeft), inRight)))) <= 0L))
                                                ^
/var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmp_81yA3.cpp:53:69: note: in instantiation of function template specialization '__pythran_bug::testFunc::operator()<long &, long &>' requested here
                                auto res = __pythran_bug::testFunc()(inLeft, inRight);
                                                                    ^
1 error generated.
In file included from /var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmp_81yA3.cpp:11:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/include/numpy/round.hpp:4:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/include/numpy/around.hpp:5:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/include/numpy/floor_divide.hpp:13:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/div.hpp:55:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/scalar/div.hpp:14:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/common/scalar/function/div.hpp:18:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/iround.hpp:44:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/scalar/iround.hpp:14:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/common/scalar/function/iround.hpp:20:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/round.hpp:57:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/simd/round.hpp:15:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/common/simd/function/round.hpp:30:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/nearbyint.hpp:43:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/simd/nearbyint.hpp:19:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/x86/sse2/simd/function/nearbyint.hpp:14:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/inearbyint.hpp:50:
In file included from /Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/function/scalar/inearbyint.hpp:21:
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/x86/sse2/scalar/function/inearbyint.hpp:31:14: error: use of undeclared identifier '_mm_cvtsd_si64'
      return _mm_cvtsd_si64(_mm_set_sd(a0));
             ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/detail/dispatch/function/functor.hpp:76:43: note: in instantiation of member function 'boost::simd::ext::impl_inearbyint_<boost::simd::sse2_, boost::dispatch::scalar_<boost::dispatch::double_<double> > >::operator()' requested here
      return BOOST_DISPATCH_IMPL_TAG_CALL(Tag,Site,Args,args);
                                          ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/numpy/rint.hpp:20:14: note: in instantiation of function template specialization 'boost::dispatch::functor<boost::simd::tag::inearbyint_, boost::simd::sse4_1_>::operator()<const double &>' requested here
      return boost::simd::inearbyint(v);
             ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/types/numpy_nary_expr.hpp:27:12: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::numpy::wrapper::rint<double>' requested here
    return NUMPY_NARY_FUNC_SYM(std::forward<T>(args)...);
           ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/numpy/rint.hpp:30:38: note: expanded from macro 'NUMPY_NARY_FUNC_SYM'
#define NUMPY_NARY_FUNC_SYM wrapper::rint
                                     ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/numpy/around.hpp:20:12: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::numpy::functor::rint::operator()<const double &>' requested here
    return functor::rint{}(a);
           ^
/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/pythonic/numpy/around.hpp:56:35: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::numpy::around<double>' requested here
  DEFINE_FUNCTOR(pythonic::numpy, around);
                                  ^
/var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmp_81yA3.cpp:39:49: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::numpy::functor::around::operator()<double>' requested here
    if ((pythonic::__builtin__::functor::int_{}(pythonic::numpy::functor::round{}((pythonic::operator_::div((1.0 * inLeft), inRight)))) <= 0L))
                                                ^
/var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmp_81yA3.cpp:53:69: note: in instantiation of function template specialization '__pythran_bug::testFunc::operator()<long &, long &>' requested here
                                auto res = __pythran_bug::testFunc()(inLeft, inRight);
                                                                    ^
1 error generated.
WARNING  Compilation error, trying hard to find its origin...
WARNING  Nop, I'm going to flood you with C++ errors!
CRITICAL Cover me Jack. Jack? Jaaaaack!!!!
E: error: Command "c++ -fno-strict-aliasing -fno-common -dynamic -arch x86_64 -arch i386 -g -Os -pipe -fno-common -fno-strict-aliasing -fwrapv -DENABLE_DTRACE -DMACOSX -DNDEBUG -Wall -Wshorten-64-to-32 -DNDEBUG -g -fwrapv -Os -Wall -Wstrict-prototypes -DENABLE_DTRACE -arch x86_64 -arch i386 -pipe -DUSE_GMP -DENABLE_PYTHON_MODULE -D__PYTHRAN__=2 -DPYTHRAN_BLAS_BLAS -I/usr/local/Cellar/openblas/0.2.20/include -I/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran -I/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/numpy/core/include -I/Users/jlaroche/temp/PythranBugs/JeanFOO/lib/python2.7/site-packages/numpy/core/include -I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c /var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmp_81yA3.cpp -o /var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmp4rAOop/var/folders/j6/7t8sg1vj4q97zhh9z5cdmxbm4rz935/T/tmp_81yA3.o -std=c++11 -fno-math-errno -w -fvisibility=hidden -fno-wrapv" failed with exit status 1`

**Comment by jeanlaroche** [comment]
I tried again with the latest version pulled from git. Still getting the issue:
Log is attached.
```import numpy as np

# pythran export testFunc(int,int)
def testFunc(inLeft, inRight):
    # nHop = int(round(1. * inLeft / inRight)) # This works
    nHop = int(np.round(1. * inLeft / inRight)) # This does NOT work
    # nHop = np.round(1. * inLeft / inRight).astype(int) # This does NOT work
    if nHop <= 0:   # <- THIS is where it crashes depending on ^^
        return 0    # <- THIS is where it crashes depending on ^^
    return 1
```
[log.txt](https://github.com/serge-sans-paille/pythran/files/1876431/log.txt)

**Comment by serge-sans-paille** [comment]
note:

```
 nHop = np.round(1. * inLeft / inRight).astype(int) # This does NOT work
```
This one does not make sense (the result of np.round is a scalar, it does not have an ``astype`` member

**Comment by serge-sans-paille** [comment]
branch fix/nearbyint may solve your issue, can you check?

**Comment by jeanlaroche** [comment]
> This one does not make sense (the result of np.round is a scalar, it does not have an astype member

Well, apparently yes:
``` import numpy as np
A = np.round(3./4.)
type(A) -> <type 'numpy.float64'>
A.astype(int) -> 1
```

**Comment by jeanlaroche** [comment]
branch fix/nearbyint does not solve my problem. 
[log.txt](https://github.com/serge-sans-paille/pythran/files/1877792/log.txt)

**Comment by jeanlaroche** [comment]
BTW, j’ai fini de compiler ma grosse fonction avec pythran, et au final ca marche! J’ai maintenant une bonne implementation en C++. Elle est un peu plus lente que ce que j’avais avec python+numba, mais j’ai pas du tout cherche a optimizer pour l’instant.
Mais ca n’a pas ete de tout repos. Pleins de problemes et toujours tres difficile de trouver quelle est l’origine de l’erreur. Il faudrait trouver un moyen de lier la ligne de code C++ qui cree l’erreur a la ligne de python qui l’a generee mais j’imagine que tu as essaye et que c’est pas simple du tout.

J

**Comment by jeanlaroche** [comment]
This still fails with master, on macos.

> /Users/jlaroche/Library/Python/2.7/lib/python/site-packages/pythran-0.8.4.post0-py2.7.egg/pythran/boost/simd/arch/x86/sse2/scalar/function/inearbyint.hpp:31:14: error: use of undeclared identifier '_mm_cvtsd_si64'
      return _mm_cvtsd_si64(_mm_set_sd(a0));
             ^

**Comment by jeanlaroche** [comment]
Update: this is now fixed if removing the -arch i386 flags (repeated twice apparently) during the compilation.

**Comment by jeanlaroche** [comment]
Fixed with b966fe6c6aba34474880e2a086373cf6ee91345e
